File: projects/GPL-FH-Java/features/UndirectedWithNeighbors/GPL/Vertex.java
package GPL;

import java.util.LinkedList;
import java.util.Iterator;

  // **********************************************************************   
   
public class Vertex
    {
    public LinkedList adjacentNeighbors;
    public String name;

    public Vertex() 
    {
        VertexConstructor();
    }
      
    public void VertexConstructor() 
    {
        name      = null;
        adjacentNeighbors = new LinkedList();
    }

    public  Vertex assignName( String name ) 
    {
        this.name = name;
        return ( Vertex ) this;
    }

    public String getName( )
    {
        return this.name;
    }
    
    public LinkedList getNeighborsObj( )
    {
 	  return adjacentNeighbors;
    }

    public VertexIter getNeighbors( )
    {
        return new VertexIter( )
        {
            private Iterator iter = adjacentNeighbors.iterator( );
            public Vertex next( ) 
            { 
                return ( ( Neighbor )iter.next( ) ).neighbor; 
            }
            public boolean hasNext( ) 
            { 
                return iter.hasNext( ); 
            }
        };
    }

    public void display( ) 
    {
        System.out.print( "Node " + name + " connected to: " );

        for ( VertexIter vxiter = getNeighbors( ); vxiter.hasNext( ); )
        {
            System.out.print( vxiter.next( ).getName( ) + ", " );
        }

        System.out.println();
    }
//--------------------
// differences
//--------------------

    public void addEdge( Neighbor n ) 
    {
        adjacentNeighbors.add( n );
    }

    public void adjustAdorns( Neighbor sourceNeighbor )
    {
    }

    public EdgeIter getEdges( )
    {
        return new EdgeIter( )
        {
            private Iterator iter = adjacentNeighbors.iterator( );
            public EdgeIfc next( ) 
            { 
                return ( Neighbor ) iter.next( ); 

//              return ( ( EdgeIfc ) ( ( Neighbor )iter.next( ) ).edge );
            }
            public boolean hasNext( ) 
            { 
              return iter.hasNext( ); 
            }
        };
    }


}

File: plugins/de.ovgu.featureide.examples/featureide_examples/GPL-FH-CSharp/features/UndirectedWithNeighbors/Vertex.cs
//layer Undirected;

using System.Collections;
using System.Collections.Generic;

  // **********************************************************************   
   
public class Vertex
    {
    public List<Neighbor> adjacentNeighbors;
    public string name;

    public Vertex() 
    {
        VertexConstructor();
    }
      
    public void VertexConstructor() 
    {
        name      = null;
        adjacentNeighbors = new List<Neighbor>();
    }

    public  Vertex assignName( string name ) 
    {
        this.name = name;
        return ( Vertex ) this;
    }

    public string GetName( )
    {
        return this.name;
    }
    
    public List<Neighbor> GetNeighborsObj( )
    {
 	  return adjacentNeighbors;
    }

    public Iterator<Vertex> GetNeighbors( )
    {
        return new Iterator<Vertex, Neighbor>(adjacentNeighbors.GetEnumerator(), TransformToNeighbor);
    }
	public Vertex TransformToNeighbor (Neighbor val){
		return val.neighbor;
	}
    public void display( ) 
    {
        System.Console.Out.Write( "Node " + name + " connected to: " );

        for ( Iterator<Vertex> vxiter = GetNeighbors( ); vxiter.hasNext( ); )
        {
            System.Console.Out.Write( vxiter.next( ).GetName( ) + ", " );
        }

        System.Console.Out.WriteLine();
    }
//--------------------
// differences
//--------------------

    public void AddEdge( Neighbor n ) 
    {
        adjacentNeighbors.Add( n );
    }

    public void adjustAdorns( Neighbor sourceNeighbor )
    {
    }

    public Iterator < EdgeIfc > GetEdges (){
	    return new Iterator < EdgeIfc, Neighbor > (adjacentNeighbors.GetEnumerator (), CastNeighborToEdgeIfc);
    }
    public static EdgeIfc CastNeighborToEdgeIfc(Neighbor val) {
        return val; //implicit upcast
    }
}
